### PR TITLE
Export yieldM on ConduitM.

### PR DESCRIPTION
**Pull request by tmcgilchrist**
So I can write normal pure functions and mix them into conduit at the last moment. @charleso 

eg
```haskell
deriveFeatures :: Monad m =>  Conduit Row (EitherT RowError m) OtherRow
deriveFeatures od = awaitForever (yieldM . hoistEither . deriveFeatures')

deriveFeatures' :: Row -> Either JoinOfferError OtherRow
deriveFeatures' = ....

```

https://github.com/snoyberg/conduit/pull/270

### CODE REVIEW

File: x-conduit/src/X/Data/Conduit.hs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+module X.Data.Conduit (
+    yieldM
+  ) where
+
+
+import           Control.Monad.Trans.Class (lift)
+import           Data.Conduit (yield, ConduitM(..))
+
+yieldM :: Monad m => m o -> ConduitM i o m ()
+yieldM mo = lift mo >>= yield
```
Comment (charleso) [inline x-conduit/src/X/Data/Conduit.hs]: Not sure if you want to add a comment about the two `yieldM` versions in conduit?

**Comment by charleso** [comment]
👍  from me, but I don't use conduit so would be tempted to ping someone else who does.

File: x-conduit/src/X/Data/Conduit.hs (hunk 1 above)
Comment (tmcgilchrist) [inline x-conduit/src/X/Data/Conduit.hs]: So conduit exposes another `yieldM` function of `yieldM :: Monad m => m o -> Pipe l i o u m ()` which uses the `Pipe` type and doesn't work seamlessly with the other functions exported by `Data.Conduit` which all expect `ConduitM` or one of it's aliases.

**Comment by domdere** [comment]
maybe also comment on when you might want to use `awaitForever` + `yieldM` over just [`mapM`](https://hackage.haskell.org/package/conduit-1.2.6.6/docs/Data-Conduit-List.html#v:mapM)

**Comment by tmcgilchrist** [comment]
`mapM` does indeed do the same thing.
I suppose I start `conduit` code with a series of stages like:
```haskell
source =$= do-thing =$= do-other-thing $$ sink
```
and have a function for each stage with the Conduit types laid out so I get them composing together correctly. Then try and have a pure version of each stage. 

Can close this if it's not helpful.

**Comment by erikd-ambiata** [comment]
I'm a Conduit user and if this was available, I'd use it as needed.

:+1: on :green_apple:

**Comment by tmcgilchrist** [comment]
Don't care anymore about this, closing.

**Pull request closed without merging**